accelerator/server.cc

```diff
@@ -466,7 +466,7 @@ int main(int argc, char* argv[]) {
         char* json_result = NULL;
 
         ret = api_get_ta_info(&json_result, &ta_core.info, &ta_core.iconf, &ta_core.cache, &ta_core.service);
-
+        ret = set_response_content(ret, &json_result);
```

Comment:
Why translating (status_t)ret to MHD_HTTP_StatusCode then assign it back to (status_t)ret.
Is it better to return status_t in those functions in http.c?

Reply:
`server.cc` and `http.c` use different framework.

Reply:
In http.c
`ta_get_url_parameter` is also return status_t and `set_response_content` is also for TA?
Expect the return value of `set_response_content`  with MHD_HTTP_StatusCode type is always matching status_t may be dangerous.

Reply:
But other places in server.cc have this logic too.
It might be discussed in another issue.

Reply:
After discussing with @HowJMay, It may need another function to convert MHD error code to TA error code. Currently,  set_response_content only be called in the server.cc and the server.cc will expire. We need to take notice of it when using it in the future.

Reply:
Before performing `set_response_content`, shall we check the return value of `api_get_ta_info`?

Reply:
No, we don't need to. `set_response_content` is responsible to choose the right HTTP error code depending on the `status_t` returned by `api_get_ta_info `

Reply:
Actually the return value of `api_get_ta_info` is wrong. It will return SC_TA_OOM regardless of any error. It the return value of `api_get_ta_info` should inherit from `ta_get_info_serialize`.

Reply:
@locke283 & @HowJMay, please create a new issue regarding error handling.